Mathlib/CategoryTheory/Grothendieck.lean

```diff
@@ -148,10 +158,55 @@ theorem congr {X Y : Grothendieck F} {f g : X ⟶ Y} (h : f = g) :
   dsimp
   simp
 
+@[simp]
+theorem base_eqToHom {X Y : Grothendieck F} (h : X = Y) :
+    (eqToHom h).base = eqToHom (congrArg Grothendieck.base h) := by subst h; rfl
+
+@[simp]
+theorem fiber_eqToHom {X Y : Grothendieck F} (h : X = Y) :
+    (eqToHom h).fiber = eqToHom (by subst h; simp) := by subst h; rfl
+
 lemma eqToHom_eq {X Y : Grothendieck F} (hF : X = Y) :
     eqToHom hF = { base := eqToHom (by subst hF; rfl), fiber := eqToHom (by subst hF; simp) } := by
   subst hF
   rfl
+
+section Transport
+
+/--
+If `F : C ⥤ Cat` is a functor and `t : c ⟶ d` is a morphism in `C`, then `transport` maps each
+`c`-based element of `Grothendieck F` to a `d`-based element.
+-/
+@[simp]
+def transport (x : Grothendieck F) {c : C} (t : x.base ⟶ c) : Grothendieck F :=
+  ⟨c, (F.map t).obj x.fiber⟩
+
+/--
+If `F : C ⥤ Cat` is a functor and `t : c ⟶ d` is a morphism in `C`, then `transport` maps each
+`c`-based element `x` of `Grothendieck F` to a `d`-based element `x.transport t`.
+
+`transport_hom` is the morphism `x ⟶ x.transport t` induced by `t` and the identity on fibers.
+-/
+@[simps]
+def toTransport (x : Grothendieck F) {c : C} (t : x.base ⟶ c) : x ⟶ x.transport t :=
+  ⟨t, 𝟙 _⟩
+
+/--
+If `F : C ⥤ Cat` and `x : Grothendieck F`, then every `C`-isomorphism `α : x.base ≅ c` induces
+an isomorphism between `x` and its transport along `α`
+-/
+@[simps]
+def transportIso (x : Grothendieck F) {c : C} (α : x.base ≅ c) :
```

Comment:
I would think that the following definition could be useful:
```lean
@[simps]
def isoMk {X Y : Grothendieck F} (e₁ : X.base ≅ Y.base)
    (e₂ : (F.map e₁.hom).obj X.fiber ≅ Y.fiber) :
    X ≅ Y where
  hom := ⟨e₁.hom, e₂.hom⟩
  inv := ⟨e₁.inv, (F.map e₁.inv).map e₂.inv ≫
    eqToHom (Functor.congr_obj (F.mapIso e₁).hom_inv_id X.fiber)⟩
  hom_inv_id := Grothendieck.ext _ _ (by simp) (by simp)
  inv_hom_id := Grothendieck.ext _ _ (by simp) (by
    have := Functor.congr_hom (F.mapIso e₁).inv_hom_id e₂.inv
    dsimp at this
    simp [this])
```
Indeed, `transportIso` becomes:
```lean
@[simps!]
def transportIso (x : Grothendieck F) {c : C} (α : x.base ≅ c) :
    x.transport α.hom ≅ x :=
  (isoMk α (Iso.refl _)).symm
```